frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx

```diff
@@ -60,28 +65,39 @@ const validateStep = (name: string, state: WizardState, t: TFunction) => {
 
 const getPayloads = (stepName: string, state: WizardState, t: TFunction) => {
   const { backingStorage, createStorageClass, storageClass, connectionDetails } = state;
-  const { externalStorage } = backingStorage;
+  const { externalStorage, deployment, type } = backingStorage;
 
   const isRhcs = externalStorage === OCSServiceModel.kind;
 
   const { createPayload, model, displayName } = getExternalStorage(externalStorage) || {};
 
-  if (model && displayName && stepName === StepsName(t)[Steps.BackingStorage] && !isRhcs) {
-    const systemName = getStorageSystemName(displayName);
+  if (
+    stepName === StepsName(t)[Steps.BackingStorage] &&
+    type === BackingStorageType.EXTERNAL &&
+    !isRhcs
+  ) {
+    const systemName = getExternalSSName(displayName);
     const systemKind = getStorageSystemKind(model);
     return [createSSPayload(systemKind, systemName)];
   }
 
-  if (stepName === StepsName(t)[Steps.ReviewAndCreate] && model && displayName) {
-    const systemName = getStorageSystemName(displayName);
-    const systemKind = model && getStorageSystemKind(model);
-    const secondParam = isRhcs ? connectionDetails : createStorageClass;
-    const payloads = [
-      ...createPayload(systemName, secondParam, model, storageClass.name),
-      createSSPayload(systemKind, systemName),
-    ];
-
-    return !_.isEmpty(secondParam) && payloads;
+  if (stepName === StepsName(t)[Steps.ReviewAndCreate]) {
```

Comment:
Lot of ifs and nested ifs. Code could use switch cases here and isolate the methods in a separate file for handling each step

```
switch(stepName){
case Steps.BackingStorage: handleBackingStorage( state, ....., .....)
case Steps.ReviewAndCreate: handleReviewAndCreate( state, ....., .....)
.......
```

This would make it a lot more cleaner

Reply:
There will only be two steps BackingStorage and ReviewAndCreate.
ReviewAndCreate will have more conditions I will refactor once I get to integrating OCS. Otherwise we will keep doing these reafctors in every PR.

Reply:
I think it's a pretty small change to do.. Just need to export the functions in another file and apply a switch.  Can be done in 5 mins 

Please make a note of this if you're planning to do it later.

Reply:
And this ifs are not just based on single value so that we can use switch all over. Still the functions are separate and you can understand logic easily.

Reply:
There are a lot of `ifs` because it supposed to be as per UXD.

Reply:
I see that but we need to isolate these per step basis. Something like this 
```
switch(stepName){
case Steps.BackingStorage: handleBackingStorage( state, ....., .....)
case Steps.ReviewAndCreate: handleReviewAndCreate( state, ....., .....)
.......

handleBackingStore = (.....) => {
if(!isrhcs && isExternal(type)){
...........
...........
...........
} 
}

handleReviewAndCreate = (.....) => {
if(isExternal(type)){
...
...
}

if(isMCGOnly()){
...
...
...
}
}
```

It would make it more palatable and mentally we'll be isolating the functionality per step.
We could check major refactors in the future

Reply:
There is more code , and we don't really need wrappers around backing storage but reveiw and create which will come in that PR. Also if you warp it inside functions then you will only have problem later in debugging :)
creating a function for a simplet statement is an overkill IMO 
const isExternal = `type === BackingStorageType.EXTERNAL`